File: frontend/lembretes/src/app/features/game-profile/models/game-profile-view-dto.ts
import { LevelViewDTO } from "../../level/services/level-view.dto";
import { UserSimpleViewDTO } from "../../user/models/user-view.dto";

export interface GameProfileViewDTO {
    id: string;
    user: UserSimpleViewDTO;
    level: LevelViewDTO;
    xp:number;
    moedas: number;
    streakAtual: number;
    melhorStreak: number;
    ultimoDiaAtivo: Date;
    totalConcluido: number;
    criadoEm: string;
    atualizadoEm: string;
    role: string
}

File: backend/lembrete/src/main/java/io/github/adrianovictorn/lembrete/dto/game_profile/GameProfileViewDTO.java
package io.github.adrianovictorn.lembrete.dto.game_profile;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

import io.github.adrianovictorn.lembrete.dto.level.LevelViewDTO;
import io.github.adrianovictorn.lembrete.dto.user.UserSimpleViewDTO;

public record GameProfileViewDTO(
    UUID id,
    UserSimpleViewDTO user,
    LevelViewDTO level,
    long xp,
    long moedas,
    int streakAtual,
    int melhorStreak,
    LocalDate ultimoDiaAtivo,
    long totalConcluido, 
    LocalDateTime criadoEm,
    LocalDateTime atualizadoEm

) {
}
